milestone1/docker-compose.yml

```diff
@@ -0,0 +1,24 @@
+version: "3"
+services:
+  app:
+    build: .
+    ports:
+      - "8080:8080"
+    depends_on:
+      - mysqldb
+
+  mysqldb:
+    image: mysql:5.7.33
+    ports:
+    - "3307:3306"
+    container_name: mysqldb
+    volumes:
+    - milestone1_db_data:/var/lib/mysql
+    environment:
+      - MYSQL_ROOT_PASSWORD=r00t!
+      - MYSQL_DATABASE=AppDB
+      - MYSQL_USER=appUser
+      - MYSQL_PASSWORD=appUser!
+
+volumes:
```

Comment:
We should add a volume (let's say, one docker volume mapped into one folder inside the project's scope).
This way we can easily maintain this and update what's needed

Reply:
From Docker Documentation:
"https://docs.docker.com/compose/#:~:text=Compose%20is%20a%20tool%20for,the%20services%20from%20your%20configuration"

- _Define your app’s environment with a Dockerfile so it can be reproduced anywhere._
- _Define the services that make up your app in docker-compose.yml so they can be run together in an isolated environment._

I think we should have the dockerfile separated from docker-compose so we can also have the possibility to create a docker image with maven.
If we decide to push the image to dockerhub (for example), docker-compose would only link to the image and not dockerfile, the dockerfile would exist to create the image in the first place.
What i've done goes according to the official docker documentation and it works, so i think we should stick with it.

Also i don't understand why we need a local volume pointing to an existing folder for the Server application, since we don't need to persist any specific data from Server.